ludwig/data/preprocessing.py

```diff
@@ -600,6 +503,161 @@ def preprocess_for_training(
     )
 
 
+def _preprocess_csv_for_training(
+        features,
+        data_csv=None,
+        data_train_csv=None,
+        data_validation_csv=None,
+        data_test_csv=None,
+        skip_save_processed_input=False,
+        preprocessing_params=default_preprocessing_parameters,
+        random_seed=default_random_seed
+):
+    """
+    Method to pre-process csv data
+    :param features: list of all features (input + output)
+    :param data_csv: path to the csv data
+    :param data_train_csv:  training csv data
+    :param data_validation_csv: validation csv data
+    :param data_test_csv: test csv data
+    :param skip_save_processed_input: if False, the pre-processed data is saved
+    as .hdf5 files in the same location as the csvs with the same names.
+    :param preprocessing_params: preprocessing parameters
+    :param random_seed: random seed
+    :return: training, test, validation datasets, training metadata
+    """
+    if data_csv is not None:
+        # Use data and ignore _train, _validation and _test.
+        # Also ignore data and train set metadata needs preprocessing
+        logging.info(
+            'Using full raw csv, no hdf5 and json file '
+            'with the same name have been found'
+        )
+        logging.info('Building dataset (it may take a while)')
+        data, train_set_metadata = build_dataset(
+            data_csv,
+            features,
+            preprocessing_params,
+            random_seed=random_seed
+        )
+        if not skip_save_processed_input:
+            logging.info('Writing dataset')
+            data_hdf5_fp = replace_file_extension(data_csv, 'hdf5')
+            data_utils.save_hdf5(data_hdf5_fp, data, train_set_metadata)
+            logging.info('Writing train set metadata with vocabulary')
+
+            train_set_metadata_json_fp = replace_file_extension(data_csv,
```

Comment:
Nit: same as the comment below.
We should have either
x = foo(
    y,
    z
)

or 
x = foo(y,
             z)

everywhere.

I would opt for option 1 since it results narrower lines.